web3/_utils/method_formatters.py

```diff
@@ -440,6 +444,14 @@ def apply_list_to_array_formatter(formatter: Any) -> Callable[..., Any]:
     RPC.eth_getFilterLogs: filter_result_formatter,
     RPC.eth_getLogs: filter_result_formatter,
     RPC.eth_getProof: apply_formatter_if(is_not_null, proof_formatter),
+    RPC.eth_getRawTransactionByBlockHashAndIndex: apply_formatter_if(
+        is_not_null,
+        transaction_formatter,
+    ),
```

Comment:
Based on [this doc](https://playground.open-rpc.org/?uiSchema%5BappBar%5D%5Bui:splitView%5D=false&schemaUrl=https://raw.githubusercontent.com/ethereum/eth1.0-specs/master/json-rpc/spec.json&uiSchema%5BappBar%5D%5Bui:input%5D=false) it looks like this method will return the hex encoded bytestring, so instead of applying the `transaction_formatter`, I think this can just become `HexBytes`, like on line 455.
```suggestion
    RPC.eth_getRawTransactionByBlockHashAndIndex: HexBytes,
```